### ISSUE TITLE
Cooperation to get anonymized user data as app developer

### ISSUE THREAD

**Issue by christianlupus** [te-online/timemanager#101]
Hello,

I am an app developer (of the cookbook app) and I recently opened this [discussion on the forum](https://help.nextcloud.com/t/generic-domain-for-app-development/147153). The idea was to provide a central domain for all app developers as a starting point. One could create sub-domains for each app to be used for documentation and other, more app-specific tasks.

One such issue was my request to have the option to collect some statistics on the installed versions. This triggered a significant push-back by some users that fear that their data is leaked out. They fear that the valuable name of Nextcloud as independent and not collecting any personal data might be at stake.

I personally think this is not too critical as I would like to collect only anonymized data and any app can do so once installed. It is just a matter of willingness.

Long story short: I would like to get the opinion on this by some other devs that might have their own agenda and pain points. Maybe this could be useful for you as well.

I will post this in a few repos in order to trigger a bit of honest discussion about the pros and cons.
Christian

**Comment by te-online** [comment]
Hey @christianlupus 👋 

Thanks for opening a discussion about this topic.

Generally, I'm not opposed to collecting data to improve especially FOSS application, as such data collection directly benefits users and isn't designed to support a company bottom line. I would be strongly opposed to metrics collection for other purposes, such as advertisement, profile building, click or referrer tracking etc.

However, I also have to say that I'm a bit tired of data collection in general. I don't feel like I need to collect anonymous metrics from random users to improve this app.

I also respect that people are using free and opensource software _precisely because_ they don't want their behavior be tracked or analyzed in any way – no matter if anonymized or not. We can call this data minimalism.

Thinking of my background in UX, I'd much rather see regular user testing being done. Create a panel of users and have them test your app. Watch them as they use your app for the first time, or how their workflow is when they're already power users.

A user test with just a handful of participants can uncover lots of UX problems in your application. More than you might have caught even with exhaustive data collection.

----

Can you generally elaborate on what you'd like to track? From your issue description it sounds like just the installed version of an app? (I'd assume most users will use the latest version anyways...)

----

Somewhat separately from data collection, I think that providing a subdomain for each app is a brilliant idea. Though, it doesn't necessarily need to be a subdomain in my view, it could just be an extra page on the app store or similar. A place to provide some more content for an app. Improving the app store listing, ratings etc. could receive higher priority, too.

----

To conclude, while I applaud the idea of improving the situation for app developers in any way, I'm not keen on implementing any kind of user tracking in my app(s) at this point.

**Comment by christianlupus** [comment]
Hello and sorry for the late reply.

As I wrote in the forum once more, this topic is split into multiple parts. The main question was the thing with a community domain for all app devs. That way every app dev (team) could decide on what they are doing with it. This might especially be tailored down to the functionality of the app in question.
As you said, this might be seen as an extension to the app store but without its restrictions (sort of). You could add supporting links or something similar with the per-app domain.

---

So far this was the main idea. However, soon the idea came up to use the domain to call home from the frontend/backend. This has all the security and privacy concerns involved that were expressed loudly in the forums.

I am claiming the adage _with great power comes great responsibility. Currently, all devs could do this in their apps in general. It might not be known/expected/made obvious to the public. One example is the password app. They have some data collected about the servers of their users. But this is technical data, not user tracking. I know this is only semantics but still another level.

Adding a domain on top of this just raises awareness of the topic. I have to admit, it might lead to a bit less skepticism by the users if the domain has an official character.

---

To conclude as well :wink:: Yes, the options are there to add tracking to the apps. It is up to the devs to decide on what is possible and take the responsibility for that.

**Comment by te-online** [comment]
Thanks for the discussion. I'm closing this for now. Feel free to post or re-open if you have new information.
